scripts/zkai_attention_kv_stwo_native_masked_sequence_proof_input.py

```
@@ -0,0 +1,527 @@
+        vector = row[vector_field]
+        if not isinstance(vector, list) or len(vector) != width:
+            raise AttentionKvStwoNativeInputError(f"{vector_field} width mismatch")
+        for index, value in enumerate(vector):
+            require_int(value, f"{vector_field}[{index}]")
+    expected_products = [left * right for left, right in zip(row["query"], row["key"], strict=True)]
+    if row["products"] != expected_products:
+        raise AttentionKvStwoNativeInputError("score product row drift")
+    if row["score"] != sum(expected_products):
+        raise AttentionKvStwoNativeInputError("score sum drift")
+    if row["score_gap"] != row["selected_score"] - row["score"] or row["score_gap"] < 0:
+        raise AttentionKvStwoNativeInputError("selected-score dominance gap drift")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)